src/components/public/swiper/CustomSwiper.tsx

```diff
@@ -0,0 +1,84 @@
+import { useRef, useState } from 'react';
+import { Swiper, SwiperSlide, SwiperClass } from 'swiper/react';
+import { Navigation, Pagination, Scrollbar } from 'swiper/modules';
+import 'swiper/css';
+import 'swiper/css/navigation';
+import 'swiper/css/scrollbar';
+
+import { ReactComponent as ArrowRight } from '../../../assets/icons/arrow-right.svg';
+import { ReactComponent as ArrowLeft } from '../../../assets/icons/arrow-left.svg';
+
+interface CustomSwiperProps<T> {
+    items: T[] | null;
+    renderItem: (item: T, index: number) => React.ReactNode;
+    slidesPerView?: number;
+    breakpoints?: Record<number, { slidesPerView: number }>;
+    showScrollbar?: boolean;
+}
+
+export function CustomSwiper<T>({
+    items,
+    renderItem,
+    slidesPerView = 1,
+    breakpoints = {},
+    showScrollbar = false,
+}: CustomSwiperProps<T>) {
+    const swiperRef = useRef<SwiperClass | null>(null);
+    const [canGoPrev, setCanGoPrev] = useState(false);
+    const [canGoNext, setCanGoNext] = useState(true);
+    const [showButtons, setShowButtons] = useState(false);
+
+    const handlePrev = () => swiperRef.current?.slidePrev();
+    const handleNext = () => swiperRef.current?.slideNext();
+
+    const updateState = (swiper: SwiperClass) => {
+        const perView = typeof swiper.params.slidesPerView === 'number' ? swiper.params.slidesPerView : 1;
+        const total = swiper.slides.length;
+        setCanGoPrev(!swiper.isBeginning);
+        setCanGoNext(!swiper.isEnd);
+        setShowButtons(total > perView);
+    };
+
+    return (
+        <>
+            {items && items.length > 0 ? (
+                <>
+                    <Swiper
+                        modules={[Navigation, Pagination, ...(showScrollbar ? [Scrollbar] : [])]}
+                        onSwiper={(swiper: SwiperClass) => {
+                            swiperRef.current = swiper;
+                            updateState(swiper);
+                            swiper.on('slideChange', updateState);
+                            swiper.on('resize', updateState);
+                            swiper.on('reachBeginning', updateState);
+                            swiper.on('reachEnd', updateState);
+                            swiper.on('fromEdge', updateState);
+                        }}
```

Comment:
You can manage the button state much more efficiently by removing the manual `updateState` function entirely. Instead, let's use Swiper's dedicated declarative props (`onInit`, `onReachBeginning`, `onReachEnd`, `onFromEdge`) to manage state. This ensures we only re-render when the slider actually hits an edge.
```suggestion
                        modules={swiperModules}
                        onInit={handleInit}
                        onResize={handleResize}
                        onReachBeginning={handleReachBeginning}
                        onReachEnd={handleReachEnd}
                        onFromEdge={handleFromEdge}
```